core/src/main/java/software/amazon/awssdk/http/pipeline/stages/MoveParametersToBodyStage.java

```diff
@@ -22,30 +22,31 @@
 import java.nio.charset.StandardCharsets;
 import software.amazon.awssdk.RequestExecutionContext;
 import software.amazon.awssdk.annotation.ReviewBeforeRelease;
+import software.amazon.awssdk.annotation.SdkProtectedApi;
 import software.amazon.awssdk.http.SdkHttpFullRequest;
 import software.amazon.awssdk.http.SdkHttpMethod;
 import software.amazon.awssdk.http.pipeline.MutableRequestToRequestPipeline;
 import software.amazon.awssdk.interceptor.AwsExecutionAttributes;
-import software.amazon.awssdk.util.SdkHttpUtils;
+import software.amazon.awssdk.utils.http.SdkHttpUtils;
 
 @ReviewBeforeRelease("Might only need to do this for certain protocols - ie query?")
 // TODO how is this going to work with streaming input posts in asyncland
 public final class MoveParametersToBodyStage implements MutableRequestToRequestPipeline {
     @Override
     public SdkHttpFullRequest.Builder execute(SdkHttpFullRequest.Builder input, RequestExecutionContext context) {
         if (shouldPutParamsInBody(input, context)) {
-            return putParams(input);
+            return changeQueryParametersToFormData(input);
         }
         return input;
     }
 
     private boolean shouldPutParamsInBody(SdkHttpFullRequest.Builder input,
                                           RequestExecutionContext context) {
         return notSimpleDb(context) &&
-               input.getHttpMethod() == SdkHttpMethod.POST &&
-               input.getContent() == null &&
-               input.getParameters() != null &&
-               input.getParameters().size() > 0;
+               input.method() == SdkHttpMethod.POST &&
+               input.content() == null &&
```

Comment:
Is content an optional now? Do we need to check isPresent?

Reply:
It's not optional in the builder. It's whatever they set in the set method. That's to avoid everything in the "get" from builders having to be optional. LMK if you think it should be the other way.

Reply:
+1, didn't realize this was on the builder.